src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel.cs

```diff
@@ -318,10 +320,11 @@ private async ValueTask ComputeViewStateAsync(CancellationToken cancellationToke
                 newItems: newViewModelItems);
         }
 
-        // Now create an interval tree out of the view models.  This will allow us to easily find the intersecting
-        // view models given any position in the file with any particular text snapshot.
-        var intervalTree = SimpleIntervalTree.Create(new IntervalIntrospector(), Array.Empty<DocumentSymbolDataViewModel>());
-        AddToIntervalTree(newViewModelItems);
+        // Now create an interval tree out of the view models.  This will allow us to easily find the intersecting view
+        // models given any position in the file with any particular text snapshot.
+        using var _ = SegmentedListPool.GetPooledList<DocumentSymbolDataViewModel>(out var models);
+        AddAllModels(newViewModelItems, models);
+        var intervalTree = BinaryIntervalTree.Create(new IntervalIntrospector(), models);
```

Comment:
tree construction at end.  this will move to the FlatArrayIntervalTree in next pr.